### PR TITLE
Retention policy rpc for existing dlq topics

### PR DESCRIPTION
**Pull request by sakshi-goyal-razorpay**
# Description
Bulk endpoint to set up retention policy on existing dlq topics
Dependent on: https://github.com/razorpay/metro/pull/421

Fixes https://razorpay.atlassian.net/browse/METRO-151

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
- [x] Tested on local, and stage by setting up retention policy on existing dlq topics

# Is it a breaking change?
No

# Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added unit tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added integration tests for the new feature (if applicable)
- [x] I have manually tested my code to the best of my abilities.

### CODE REVIEW

File: pkg/messagebroker/kafka_test.go [hunk 1]
```diff
@@ -43,11 +43,30 @@ func Test_Pause(t *testing.T) {
...
+	a := getKafkaAdmin()
+	resp, err := a.DescribeTopicConfigs(context.Background(), []string{})
+	assert.NotNil(t, err)
+	assert.Nil(t, resp)
+}
+
+func Test_AlterTopicConfigs(t *testing.T) {
+	a := getKafkaAdmin()
+	resp, err := a.AlterTopicConfigs(context.Background(), ModifyTopicConfigRequest{TopicConfigs: nil})
+	assert.NotNil(t, err)
```
Comment (akashgoyal96) [inline pkg/messagebroker/kafka_test.go:56]: @sakshi-goyal-razorpay we are asserting not nil for err, should we also add a nil test case as well?

Comment (sakshi-goyal-razorpay) [reply pkg/messagebroker/kafka_test.go:56]: @akashgoyal96 Its a unit test. It won't actually be able to get to kafka broker to alter or describe configs. Those cases are added as part of integration test

File: service/web/adminserver.go [hunk 2]
```diff
@@ -254,6 +254,15 @@ func (s adminServer) MigrateSubscriptions(ctx context.Context, subscriptions *me
 	return &emptypb.Empty{}, nil
 }
 
+func (s adminServer) SetupRetentionPolicy(ctx context.Context, _ *emptypb.Empty) (*metrov1.Topics, error) {
+	logger.Ctx(ctx).Infow("received request to set up retention policy")
```
Comment (akashgoyal96) [inline service/web/adminserver.go:258]: Is this log required, if yes we should add more details like topic

**Review by akashgoyal96 — COMMENTED** [review]
@sakshi-goyal-razorpay I can see Setup/Alter/describe retention policy, is execution of retention done in already done in some other PR?

File: service/web/adminserver.go (hunk 2 above)
Comment (sakshi-goyal-razorpay) [reply service/web/adminserver.go:258]: @akashgoyal96 Request object doesn't have any topic parameter. We set up retention policy on all the dlq topics and return the updated topics in response

**Comment by sakshi-goyal-razorpay** [comment]
> @sakshi-goyal-razorpay I can see Setup/Alter/describe retention policy, is execution of retention done in already done in some other PR?
we just need to set up retention, kafka will take care of the rest.

**Review by akashgoyal96 — APPROVED** [review]
LGTM

**Review by subintp — APPROVED** [review]

File: internal/topic/core_test.go [hunk 3]
```diff
@@ -547,3 +547,64 @@ func TestCore_Get(t *testing.T) {
...
+		existingConfigs map[string]map[string]string
+		expected        []string
+		listErr         error
+		getAdminErr     error
+		expectedErr     error
+	}{
+		{
+			name:     "Alter retention configs without errors",
+			topics:   []common.IModel{dlqTopic},
+			expected: []string{dlqTopic.Name},
```
Comment (subintp) [inline internal/topic/core_test.go:571]: Can you add a test case with 4 topics and 1 dlq topic. We need to assert the the AlterTopicConfigs is called only with dlq topic with retention changed.

**Review by subintp — APPROVED** [review]

**Review by subintp — CHANGES_REQUESTED** [review]
Add one more test mentioned in the comment

**Review by subintp — APPROVED** [review]

**Pull request merged**